controller/resource.go

```diff
@@ -110,6 +110,9 @@ type Task struct {
 }
 
 // ID returns Task definition ARN or error if it cannot be determined from the Task.
+// We're using task definition ARN instead of the Task ARN because we consider tasks
+// for the same definition to be equivalent in terms of their Upsert and Delete behavior,
+// and so there's no reason to process subsequent tasks with the same task definition.
```

Comment:
@lkysow I've added clarifying comment here instead. 

I definitely see what you mean though. On the one hand, I'd expect the Task's ID to be task ARN. On the other, I think a task could have a different ID depending on what makes sense for your implementation, and in this case, we can think of it as tasks with the same definition are equal to each other.

I'm a bit hesitant about the rename you suggested because I find it a bit more confusing than this for two reasons:
1. If rename `TaskLister` to `TaskDefinitionLister` it's be inconsistent in that it's actually listing tasks. I know you've mentioned that we can help clarify that with a comment, but that's where the second reason comes in.
2. The `Task` resource is a wrapper around `ecs.Task` struct, and so if we rename it to be `TaskDefinition` instead but it'll still have a reference to `ecs.Task`, I think it'll also be confusing.

Also, happy to hear feedback from @pglass to see what makes more sense. I could have been spending too much time with this code and just have tunnel vision.

Reply:
Yeah, I think I'm fine with it as-is.

Is there another way for the controller to determine if something is a mesh-task, so that we don't have to fetch tags 🤔 ? Then we could get away with using `ListTaskDefinitions` and everything would be a littler nicer.

Reply:
I actually realized that we can't list task definitions by cluster, so I don't think we can use it.